org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/DocumentSymbolHandler.java

```diff
@@ -228,7 +202,119 @@ private boolean isSyntheticElement(IJavaElement element) {
 		}
 	}
 
-	public static SymbolKind mapKind(IJavaElement element) {
+	/**
+	 * Returns with the human readable name of the element. For types with type
+	 * arguments, it is {@code Comparable<T>} instead of {@code Comparable}. First,
+	 * this method tries to retrieve the
+	 * {@link JavaElementLabels#getElementLabel(IJavaElement, long) label} of the
+	 * element, then falls back to {@link IJavaElement#getElementName() element
+	 * name}. Returns {@code null} if the argument does not have a name.
+	 */
+	public static String getName(IJavaElement element) {
+		Assert.isNotNull(element, "element");
+		String name = JavaElementLabels.getElementLabel(element, ALL_DEFAULT);
+		return name == null ? element.getElementName() : name;
+	}
+
+	/**
+	 * The location, encapsulating the range enclosing this symbol not including
+	 * leading/trailing whitespace but everything else like comments.
+	 */
+	public static Location getFullLocation(IJavaElement element) throws JavaModelException {

```

Comment:
similar static methods related to location computation live in JDTUtils. I reckon it's getting bloated over time, so we might get a dedicated LocationUtils class to handle all that